cockatrice/src/shortcutssettings.cpp

```diff
@@ -0,0 +1,248 @@
+    shortCutsFile.beginGroup("Defaults");
+    shortCutsFile.setValue(name, stringifySecuence(defaultShortCut));
+    shortCutsFile.endGroup();
+    return defaultShortCut;
+}
+
+QList<QKeySequence> ShortcutsSettings::getShortcut(QString name, QKeySequence defaultShortCut)
+{
+    return getShortcut(name, QList<QKeySequence>() << defaultShortCut);
+}
+
+QKeySequence ShortcutsSettings::getSingleShortcut(QString name, QKeySequence defaultShortCut)
+{
+    return getShortcut(name,defaultShortCut).at(0);
+}
+
+QString ShortcutsSettings::getDefaultShortcutString(QString name)
+{
+    return stringifySecuence(defaultShortCuts.value(name));
+}
+
+QString ShortcutsSettings::getShortcutString(QString name)
+{
+    return stringifySecuence(shortCuts.value(name));
+}
+
+QString ShortcutsSettings::stringifySecuence(QList<QKeySequence> secuence) const
+{
+    QString stringSecuence;
+    for(int i=0; i < secuence.size(); ++i)
+    {
+        stringSecuence.append(secuence.at(i).toString(QKeySequence::PortableText));
+        if(i < secuence.size() - 1)
+            stringSecuence.append(";");
+    }
+
+    return stringSecuence;
+}
+
+QList<QKeySequence> ShortcutsSettings::parseSecuenceString(QString stringSecuence)
+{
+    QStringList secuences = stringSecuence.split(";");
+    QList<QKeySequence> secuenceList;
+    for(QStringList::const_iterator ss = secuences.constBegin(); ss != secuences.constEnd(); ++ss)
+    {
+        secuenceList.append(QKeySequence(*ss, QKeySequence::PortableText));
+    }
+
+    return secuenceList;
+}
+
+void ShortcutsSettings::setShortcuts(QString name, QList<QKeySequence> secuence)
+{
+    shortCuts[name] = secuence;
+
+    QSettings shortCutsFile(settingsFilePath, QSettings::IniFormat);
+    shortCutsFile.beginGroup("Custom");
+    QString stringSecuence = stringifySecuence(secuence);
+    shortCutsFile.setValue(name, stringSecuence);
+    shortCutsFile.endGroup();
+}
+
+void ShortcutsSettings::setShortcuts(QString name, QKeySequence secuence)
+{
+    setShortcuts(name, QList<QKeySequence>() << secuence);
+}
+
+void ShortcutsSettings::setShortcuts(QString name, QString secuences)
+{
+    setShortcuts(name,parseSecuenceString(secuences));
+}
+
+bool ShortcutsSettings::isValid(QString name, QString secuences)
+{
+    QString checkKey = name.left(name.indexOf("/"));
+
+    QStringList stringSecuences = secuences.split(";");
+
+    QList<QString> allKeys = shortCuts.keys();
+    for(int i=0; i < allKeys.size(); i++){
+        QString key = allKeys.at(i);
+        if(key.startsWith(checkKey) || key.startsWith("MainWindow") || checkKey.startsWith("MainWindow"))
+        {
+            QString storedSecuence = stringifySecuence(shortCuts.value(key));
+            for(int j = 0; j < stringSecuences.size(); j++)
+            {
+                if(storedSecuence.contains(stringSecuences.at(j)))
+                    return false;
+            }
+        }
+    }
+    return true;
+}
+
+void ShortcutsSettings::fillDefaultShorcuts()
+{
```

Comment:
I see default shortcuts are defined here, but used only when the "restore default shortcut" button is pressed, eg:
```
+    defaultShortCuts["MainWindow/aFullScreen"] = parseSecuenceString("Ctrl+F");
```
Then, in every call to ```settingsCache->shortcuts().getShortcut()``` the default shortcut is passed as the second parameter, eg:
```
+    aFullScreen->setShortcuts(settingsCache->shortcuts().getShortcut(
+                                  "MainWindow/aFullScreen", QKeySequence("Ctrl+F")));
```
Could the getShortcut() method use the defaults defined here? This would avoid the possible human error of adding/changing a default shortcut in one place forgetting to update the other place.